lib/generators/rails/serializer_generator.rb

```diff
@@ -23,12 +23,14 @@ def association_names
       end
 
       def parent_class_name
-        if options[:parent]
-          options[:parent]
-        elsif defined?(::ApplicationSerializer)
-          'ApplicationSerializer'
+        if parent = options[:parent]
+          parent
         else
-          'ActiveModel::Serializer'
+          namespace = class_path.map(&:camelize).join('::')
+          application_serializer = [namespace, 'ApplicationSerializer']
+            .reject(&:blank?).join('::')
+          path = File.join('app/serializers', class_path, "application_serializer.rb")
```

Comment:
since this is only relevant in a Rails app, I'd rather rely on Rails autoloading for a more robust support and less code on us.

i.e.

```ruby
  return parent if (parent = options[:parent])
  ApplicationSerializer.name
rescue NameError
  'ActiveModel::Serializer'
end
```